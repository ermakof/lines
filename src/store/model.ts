import IUserProfile from '@src/modules/Auth/model/IUserProfile';

export interface IState {
  isLoading?: boolean;
  gameFieldSize: number;
  gameFieldPercentFilled: number;
  gameFieldData: Array<number>;
  selectedCell?: number;
  userLevel: string;
  userProfile?: IUserProfile;
}
