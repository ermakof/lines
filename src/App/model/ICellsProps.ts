export interface ICellsProps {
  [key: string]: string;
}
