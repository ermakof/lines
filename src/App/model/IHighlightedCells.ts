export interface IHighlightedCells {
  [key: string]: string;
}
