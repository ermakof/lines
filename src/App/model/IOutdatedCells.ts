export interface IOutdatedCells {
  [key: string]: boolean;
}
