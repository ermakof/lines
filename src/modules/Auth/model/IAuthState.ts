import IUserProfile from '@src/modules/Auth/model/IUserProfile';

export interface IAuthState {
  userProfile?: IUserProfile;
}
