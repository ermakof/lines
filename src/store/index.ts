import { configureStore } from '@reduxjs/toolkit';
import logger from 'redux-logger';

import rootReducer from '@src/store/rootReducer';

const store = configureStore({
  reducer: rootReducer,
  middleware: (getDefaultMiddleware) =>
    getDefaultMiddleware()
      .prepend
      // correctly typed middlewares can just be used
      ()
      // prepend and concat calls can be chained
      .concat(logger),
});

type TRootState = ReturnType<typeof store.getState>;

export { store, TRootState };
