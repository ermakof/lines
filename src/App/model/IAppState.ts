import { IOutdatedCells } from '@src/App/model/IOutdatedCells';

export interface IAppState {
  isLoading?: boolean;
  gameFieldPercentFilled: number;
  gameFieldData: Array<number>;
  outdatedCells?: IOutdatedCells;
  score: number;
  selectedCell?: number;
  userLevel: string;
  minChainLength: number;
}
