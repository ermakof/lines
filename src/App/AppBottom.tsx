import React, { FC, memo } from 'react';
import { useSelector } from 'react-redux';

import Panel from '@src/layout/Panel';
import styled from '@emotion/styled';
import { TRootState } from '@src/store';
import Percentage from '@src/components/Percentage';
import Score from '@src/components/Score';

interface IRoot {
  active: boolean;
}
const Root = styled.div<IRoot>`
  opacity: ${({ active }) => (active ? 1 : 0.3)};
`;

const AppBottom: FC = () => {
  const { gameFieldData, score = 0 } = useSelector(({ app }: TRootState) => app);
  const isAuthorised = useSelector(({ auth }: TRootState) => !!auth.userProfile);

  return (
    <Root active={isAuthorised}>
      <Panel role="bottomPanel">
        <Score value={score} />
        <Percentage data={gameFieldData} />
      </Panel>
    </Root>
  );
};

export default memo(AppBottom);
