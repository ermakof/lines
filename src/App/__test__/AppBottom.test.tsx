import React from 'react';
import { Provider } from 'react-redux';
import { render, screen } from '@testing-library/react';
import AppBottom from '@src/App/AppBottom';
import { store } from '@src/store';
import { IAction } from '@src/model';

let dispatch: (action: IAction) => void;

describe('AppBottom', () => {
  beforeEach(() => {
    dispatch = jest.fn();
  });

  it('Render <AppBottom> without data', () => {
    const state = {
      gameLevel: '1',
      gameFieldSize: 3,
      gameFieldPercentFilled: 10,
      gameFieldData: [1, 0, 0, 0, 0, 0, 0, 0, 0],
    };
    const { asFragment } = render(
      <Provider store={store}>
        <AppBottom />
      </Provider>
    );
    expect(asFragment()).toMatchSnapshot();
    const dataMessage = screen.getByText(/Результат/gi);
    expect(dataMessage).toBeInTheDocument();
    const percentMessage = screen.getByText(/Процент заполнения/gi);
    expect(percentMessage).toBeInTheDocument();
  });
});
