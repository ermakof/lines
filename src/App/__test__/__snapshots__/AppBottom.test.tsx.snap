// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppBottom Render <AppBottom> without data 1`] = `
<DocumentFragment>
  <div
    class="css-eec57t"
  >
    <div
      class="css-1w0hfqs"
      role="bottomPanel"
    >
      <p
        class="css-o9gtt2"
        role="messageResult"
      >
        Результат: 0
      </p>
      <p
        class="css-o9gtt2"
        role="messagePercentFilled"
      >
        Процент заполнения: 0.00
      </p>
    </div>
  </div>
</DocumentFragment>
`;
